File: Java/ProtoDebugger/src/protodebugger/model/descriptors/BooleanFieldDescriptorContainer.java
<<<<<<< HEAD
package protodebugger.model.descriptors;

import org.eclipse.swt.SWT;
import org.eclipse.swt.widgets.Button;
import org.eclipse.swt.widgets.Composite;
import org.eclipse.swt.widgets.Widget;


import com.google.protobuf.Descriptors;
import com.google.protobuf.GeneratedMessage.Builder;

public class BooleanFieldDescriptorContainer extends FieldDescriptorContainer {

	private Button check;
	public BooleanFieldDescriptorContainer(Descriptors.FieldDescriptor field)
	{
		super(field);
	}
	@Override
	public Object getValue() {
		if(value != null)
			return value.toString();
		else 
			return false;
	}

	@Override
	public boolean buildMsg(Builder<?> build) {
		if(field.isRepeated())
			build.addRepeatedField(field, check.getSelection());
		else
			build.setField(field, check.getSelection());
		return true;
	}

	@Override
	public String toString() {
		return "BooleanField name = " + name;
	}

	@Override
	public Widget getWidget(Composite parent) {
		if(check == null)
		{
			check = new Button(parent, SWT.CHECK);
			if(value != null)
				check.setEnabled((Boolean)value);
		}
		else if(check.getParent() != parent)
			check.setParent(parent);
		return check;
	}

	@Override
	public Composite getParent() {
		if(check != null)
			return check.getParent();
		return null;
	}
	

}
=======
package protodebugger.model.descriptors;

import org.eclipse.swt.SWT;
import org.eclipse.swt.widgets.Button;
import org.eclipse.swt.widgets.Composite;
import org.eclipse.swt.widgets.Widget;


import com.google.protobuf.Descriptors;
import com.google.protobuf.GeneratedMessage.Builder;

public class BooleanFieldDescriptorContainer extends FieldDescriptorContainer {

	private Button check;
	public BooleanFieldDescriptorContainer(Descriptors.FieldDescriptor field)
	{
		super(field);
	}
	@Override
	public Object getValue() {
		if(value != null)
			return value.toString();
		else if(defaultValue != null)
			return defaultValue.toString();
		else 
			return false;
	}

	@Override
	public boolean buildMsg(Builder<?> build) {
		if(field.isRepeated())
			build.addRepeatedField(field, check.getSelection());
		else
			build.setField(field, check.getSelection());
		return true;
	}

	@Override
	public void setValue(Object value) {
		this.value = value;
	}

	@Override
	public String toString() {
		return "BooleanField name = " + name;
	}

	@Override
	public Widget getWidget(Composite parent) {
		if(check == null)
			check = new Button(parent, SWT.CHECK);
		else if(check.getParent() != parent)
			check.setParent(parent);
		return check;
	}

	@Override
	public Composite getParent() {
		if(check != null)
			return check.getParent();
		return null;
	}

}
>>>>>>> 5a14f9bb2eb2e5da316d0185feeade7969c536bd

File: CPP/ProtoDebugger/src/protodebugger/model/descriptors/BooleanFieldDescriptorContainer.cpp
#include "BooleanFieldDescriptorContainer.h"
#include <google/protobuf/message.h>
#include <sstream>

using namespace std;
using namespace google;
using namespace protobuf;

QWidget * BooleanFieldDescriptorContainer::getWidget(QWidget * parent)
{
    if(m_check == NULL)
    {
        m_check = new QCheckBox(parent);

        if (m_type == FieldDescriptor::CPPTYPE_BOOL)
        {
            m_check->setChecked((bool)m_value);
        }
    }
    else if (m_check->parentWidget() != parent)
    {
        m_check->setParent(parent);
    }

    return m_check;
}

QWidget * BooleanFieldDescriptorContainer::getParent()
{
    if(m_check != NULL)
    {
       return m_check->parentWidget();
    }

    return NULL;
}

Object * BooleanFieldDescriptorContainer::getValue()
{
    stringstream ss;
    if(m_value != NULL)
    {
        if (m_type == FieldDescriptor::CPPTYPE_BOOL)
        {
            ss << (bool) m_value;
            return (Object *) ss.str().c_str();
        }
     }
    else if (m_defaultValue != NULL)
    {

    }

    return (Object *) "";
}

void BooleanFieldDescriptorContainer::setValue(Object * value)
{
   this->m_value = value;
}

QString BooleanFieldDescriptorContainer::toString()
{
    string str = "BooleanField name = " + this->m_name;
    return QString(str.c_str());
}

//Note: Using it passes in a mutable message
bool BooleanFieldDescriptorContainer::buildMsg(Message * msg)
{
    if(m_field->is_repeated())
    {
        msg->GetReflection()->AddBool(msg, m_field, m_check->isChecked());
    }
    else
    {
        msg->GetReflection()->SetBool(msg, m_field, m_check->isChecked());
    }
    return true;
}
